File: packages/core/src/models/quick-phrase.ts
/**
 * A pre-translated emergency phrase for instant communication
 * without requiring AI inference.
 */
export interface QuickPhrase {
  /** Unique phrase identifier */
  id: string;
  /** Category for grouping */
  category: 'medical' | 'safety' | 'logistics' | 'identification';
  /** English canonical text */
  sourceText: string;
  /** Translations keyed by ISO 639-1 language code */
  translations: Record<string, string>;
  /** Whether this phrase is marked as a favorite */
  isFavorite: boolean;
}

export type PhraseCategory = QuickPhrase['category'];

File: android/app/src/main/java/com/firstvoice/app/data/model/QuickPhrase.kt
package com.firstvoice.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class QuickPhrase(
    val id: String,
    val category: PhraseCategory,
    val sourceText: String,
    val translations: Map<String, String>,
    val isFavorite: Boolean = false
)

@Serializable
enum class PhraseCategory {
    medical, safety, logistics, identification;

    fun displayName(): String = when (this) {
        medical -> "Medical"
        safety -> "Safety"
        logistics -> "Logistics"
        identification -> "Identification"
    }
}
